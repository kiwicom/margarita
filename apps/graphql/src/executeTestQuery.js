// @flow

import { graphql as originalGraphQL } from 'graphql';

import schema from './Schema';
import createContext from './services/GraphQLContext';

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?Object,
  acceptLanguage: ?string,
): Promise<Object> {
  return originalGraphQL(schema, query, null, createContext(), variables);
}
