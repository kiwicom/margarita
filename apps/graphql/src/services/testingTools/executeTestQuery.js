// @flow

import { graphql as originalGraphQL } from 'graphql';

import schema from '../../Schema';
import createContext from '../graphqlContext/GraphQLContext';

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?Object,
): Promise<Object> {
  return originalGraphQL(schema, query, null, createContext(), variables);
}
