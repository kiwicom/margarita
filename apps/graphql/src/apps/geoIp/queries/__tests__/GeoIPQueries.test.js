// @flow

import { generateTestsFromFixtures } from '@kiwicom/test-utils';

import executeTestQuery from '../../../../services/testingTools/executeTestQuery';

const geoIPData = {
  isoCountryCode: 'CZ',
  latitude: '50.0848',
  longitude: '14.411200000000001',
};

generateTestsFromFixtures(`${__dirname}/__fixtures__/`, input => {
  fetch.mockResponses([JSON.stringify(geoIPData)]);
  return executeTestQuery(input);
});
