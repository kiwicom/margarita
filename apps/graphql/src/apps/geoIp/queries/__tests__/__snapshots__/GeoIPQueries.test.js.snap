// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: GeoIP.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  geoIP(ip: "62.84.146.130") {
    isoCountryCode
    coordinates {
      lat
      lng
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "geoIP": {
      "isoCountryCode": "CZ",
      "coordinates": {
        "lat": 50.0848,
        "lng": 14.411200000000001
      }
    }
  }
}
`;
